modules/framework/src-js-native/RunnerCompat.scala

```diff
@@ -205,12 +207,17 @@ private[weaver] object ReadWriter {
   }
 
   def writer(f: Writer => Unit): String = {
-    val buf = ByteBuffer.allocate(2048)
+    val baos = new ByteArrayOutputStream(2048)
+    val dos  = new DataOutputStream(baos)
```

Comment:
**thought:** two words about this:

- 2048 is now the initial length of the array, but BAOS makes it grow when needed
- I tried to switch to `StringBuilder` as suggested, but the output should have constant-length encoding of numbers, and it seemed bothersome to try and pad them to make it work. Instead, `DataOutputStream` encodes the numbers in binary as expected.
- I'm not super comfortable with this encoding even in its current shape, because it assumes the bytes can be safely decoded from UTF-8 (`new String`) and encoded again (`.getBytes()`) without data loss, and I don't think that's generally true.

For example, this prints false:

```
val bb = ByteBuffer.allocate(20)
bb.putInt(42)
bb.putDouble(21.37)
val result = bb.array()
println(new String(result).getBytes().toList == result.toList)
```

so... I think ultimately the right solution would indeed be to use `StringBuilder`, albeit with a different encoding.

Thoughts?

Reply:
ah, miraculously our doubles never have the floating point part (`.toMillis.toDouble`). However, even 50 000 ms (50 seconds) is enough to fail the roundrip.

Reply:
ok, added some tests - 3a01e09bdf174b1ea909b9915e73bed15e98f590 - for the time being, encoding the bytes in base64 to bypass the UTF-8 problem, just to have a green build. Maybe it's fine.

Reply:
finally, living proof of the roundtrip problem:

```scala
package demo

import cats.effect.IO
import weaver._
import concurrent.duration._

object DemoTest extends SimpleIOSuite {

  test("this should fail rather than hang") {
    IO.unit.as(failure("verbosity goes here"))
  }
}

object DemoTest2 extends SimpleIOSuite {

  test("this should fail rather than hang") {
    IO.unit.as(failure("verbosity goes here"))
  }
}
```

0.8.3:

<img width="860" alt="image" src="https://github.com/disneystreaming/weaver-test/assets/894884/fff0d812-acd0-4a6d-9011-e6fe44ad1fac">

another run:

<img width="822" alt="image" src="https://github.com/disneystreaming/weaver-test/assets/894884/70b4e2ee-e356-4301-9fd7-d567cca1acfc">

206de20df299981816821c78f8fdf743b1c7540f :

<img width="827" alt="image" src="https://github.com/disneystreaming/weaver-test/assets/894884/a6d04522-3f7b-46df-9e1d-54338612f283">